shared/util/saga.js

```diff
@@ -54,24 +83,30 @@ function* chainAction<Actions>(
       )
     } finally {
       if (yield Effects.cancelled()) {
-        logger.info('chainAction cancelled')
+        sl.info('chainAction cancelled')
       }
     }
   })
 }
 
-function* chainGenerator<Actions>(
+function* chainGenerator<Actions: {+type: string}>(
   pattern: RS.Pattern,
-  f: (state: TypedState, action: Actions) => Generator<any, void, any>
+  f: (state: TypedState, action: Actions, logger: SagaLogger) => Generator<any, void, any>,
+  fcnTag?: string // tag for logger
 ): Generator<any, void, any> {
   type Fn = Actions => RS.Saga<void>
   return yield Effects.takeEvery<Actions, void, Fn>(pattern, function* chainActionHelper(
     action: Actions
   ): RS.Saga<void> {
+    const sl = new SagaLogger(action.type, fcnTag || 'unknown')
     try {
       const state = yield* selectState()
-      yield* f(state, action)
+      yield* f(state, action, sl)
+      if (sl.isTagged) {
+        sl.info('calling -> ok')
```

Comment:
Why is this `calling -> ok` and the other one is ` -> ok` ?

Reply:
Just messy, fixed!